index.js

```diff
@@ -1,54 +1,101 @@
-const authorInput = document.querySelector('#author');
-const bookList = document.querySelector('#book-list');
-const formGroup = document.querySelector('#form-container');
-let addOn;
-let bookObj = JSON.parse(localStorage.getItem('saveItem')) || [];
-
-// Add book
-function newAdd() {
-  addOn = {
-    title: titleInput.value,
-    author: authorInput.value,
-    id: Math.floor(Math.random() * 1000000),
-  };
-  bookObj.push(addOn);
-  localStorage.setItem('saveItem', JSON.stringify(bookObj));
+/* eslint-disable */
+// Book class
+class Book {
+  constructor(title, author) {
+    this.title = title;
+    this.author = author;
+  }
 }
 
-// Remove
-function removeBook(id) {
-  bookObj = bookObj.filter((filterObj) => filterObj.id !== id);
-  localStorage.setItem('saveItem', JSON.stringify(bookObj));
+// Add books classas
+class displayOn {
+  static displayBooks() {
+    const books = storageStore.getBooks();
+    books.forEach((bookAddOn) => displayOn.addToBooks(bookAddOn));
+  }
+
+  static addToBooks(bookAddOn) {
+    const tableDiv = document.createElement('div');
+    tableDiv.classList.add('collection');
+    tableDiv.innerHTML = `
+    <div class="collection-text">
+    <p>"${bookAddOn.title}"</p>
+    <p>by</p>
+    <p>${bookAddOn.author}</p>
+    </div>
+    <button type="submit" class="delete">Remove</button>
+    `;
+    bookList.appendChild(tableDiv);
+  }
+
+  static deleteBook(el) {
+    if (el.classList.contains('delete')) {
+      el.parentElement.remove();
+    }
+  }
+
+  static clearInputFields() {
+    document.querySelector('#title').value = '';
+    document.querySelector('#author').value = '';
+  }
 }
 
-// Create Books
-function addBooks(addOn) {
-  const tableDiv = document.createElement('div');
-  const bookTitle = document.createElement('h3');
-  const authorName = document.createElement('h3');
-  const removeBtn = document.createElement('button');
-  const line = document.createElement('hr');
-  bookTitle.innerText = addOn.title;
-  authorName.innerText = addOn.author;
-  removeBtn.innerText = 'Remove';
-  tableDiv.append(bookTitle, authorName, removeBtn, line);
-  bookList.appendChild(tableDiv);
-  removeBtn.addEventListener('click', () => {
-    removeBtn.parentElement.remove();
-    removeBook(addOn.id);
-  });
+// Store class local storage
+class storageStore {
+  static getBooks() {
+    let books;
+    if (localStorage.getItem('books') === null) {
+      books = [];
+    } else {
+      books = JSON.parse(localStorage.getItem('books'));
+    }
+    return books;
+  }
+
+  static saveBooks(book) {
+    const books = storageStore.getBooks();
+    books.push(book);
+    localStorage.setItem('books', JSON.stringify(books));
+  }
+
+  static removeBooks(title) {
+    const books = storageStore.getBooks();
+
+    books.forEach((book, index) => {
+      if (book.title !== title) {
+        books.splice(index, 1);
+      }
+    });
+    localStorage.setItem('books', JSON.stringify(books));
+  }
```

Comment:
- [ ] I noticed that the delete function is not working as expected, for example if I have two books and I delete the second book, the first book will be deleted from local storage. Kindly fix this issue 💯

https://user-images.githubusercontent.com/45231276/191465761-a293f048-2575-49a0-8048-35fea9ea5682.mp4